src/libinjector/injector_stack.c

```diff
@@ -458,7 +458,7 @@ bool setup_linux_stack(vmi_instance_t vmi, x86_registers_t* regs, struct argumen
         // 1: rdi
         // 2: rsi
         // 3: rdx
-        // 4: rcx
+        // 4: r10
```

Comment:
Imo if we make this change there should be a comment explaining why it happend (maybe link to the issue with discussion).

Reply:
This was a needed change just for this PR that's why the change is there, otherwise, I wanted a separate commit for this I am handling all those related changes in that PR